### PR TITLE
[docker] build and publish docker image

### PR DESCRIPTION
**Pull request by obviyus**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# set base image (host OS)
+FROM python:3.8
+
+# set the working directory in the container
+WORKDIR /code
+
+# copy the dependencies file to the working directory
+COPY requirements.txt .
```
Comment (Tyagdit) [inline Dockerfile:8]: Why not copy the whole directory here instead of later

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (Tyagdit) [inline .github/workflows/python.yml]: Don't need this until we write tests, probably should comment it out for now

File: .github/workflows/python.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Lint with flake8
+      run: |
```
Comment (Tyagdit) [inline .github/workflows/python.yml:25]: install flake8 here instead

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          username: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+          password: ${{ secrets.DOCKERHUB_USERNAME }}
+      -
+        name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: superserious/bot:latest
```
Comment (Tyagdit) [inline .github/workflows/docker.yml]: SHouldn't this be a branch?

Comment (obviyus) [reply .github/workflows/docker.yml]: Nope, that's the target of the Docker image that we generate. We always want the newest image to be at `superserious/bot:latest`.

File: configuration/__init__.py [hunk 5]
```diff
@@ -1,11 +1,25 @@
...
-    with open("config.yaml", 'r') as config:
-        data = yaml.safe_load(config)
-        config = data
-except FileNotFoundError:
-    raise FileNotFoundError(
-        "\nPlease create a 'config.yaml' file and put the credentials needed for the bot in it.\n"
-        "Refer configuration/example_config.yaml for a template"
-    )
+    # By default, try to look for API keys in environment variables
+    data = {}
```
Comment (Tyagdit) [inline configuration/__init__.py]: This should be "config"

File: configuration/__init__.py [hunk 6]
```diff
@@ -1,11 +1,25 @@
...
+    data["GOODREADS_API_KEY"] = os.environ["GOODREADS_API_KEY"]
+    data["AIRVISUAL_API_KEY"] = os.environ["AIRVISUAL_API_KEY"]
+    data["JOGI_FILE_ID"] = os.environ["JOGI_FILE_ID"]
+    data["FOR_WHAT_ID"] = os.environ["FOR_WHAT_ID"]
+    data["PUNYA_SONG_ID"] = os.environ["PUNYA_SONG_ID"]
+except KeyError:
+    try:
+        with open("config.yaml", 'r') as config:
+            data = yaml.safe_load(config)
+            config = data
```
Comment (Tyagdit) [inline configuration/__init__.py]: This is redundant i guess
```suggestion
            config = yaml.safe_load(config)
```

File: configuration/__init__.py [hunk 7]
```diff
@@ -1,11 +1,25 @@
...
+    except FileNotFoundError:
+        raise FileNotFoundError(
+            "\nPlease create a 'config.yaml' file and put the credentials needed for the bot in it.\n"
+            "Refer configuration/example_config.yaml for a template"
```
Comment (Tyagdit) [inline configuration/__init__.py]: Add a line for env vars here

**Review by Tyagdit — APPROVED** [review]

**Pull request merged**